### PR TITLE
feat(AAiT.web.g3): Add blog list page

### PR DESCRIPTION
**Pull request by yohannes-07**
- I have added blog list page with SearchBar, BlogCard and Pagination components. 
- The pagination uses tailwind CSS with dynamic page numbers based on the size of the blogs list 
- I have added blog.ts type file and some tailwind apply directive CSS under global.css file.

Screenshot from 2023-08-23 11-31-02

Screenshot from 2023-08-23 11-31-50

### CODE REVIEW

File: aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  onPageChange: (page: number) => void;
+}
+
+type Props = PaginationProps & React.HTMLProps<HTMLDivElement>;
+
+const Pagination: React.FC<Props> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+  className, // This will be used to apply additional class names
```
Comment (Segnicho) [inline aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx:15]: is this comment needed? can't the code be made self-explanatory.

File: aait/web/group-3/starter_project_group3/app/blogs/page.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    {
+        _id: "2",
+        name: "Jane Smith",
+        email: "jane@example.com",
+        image: "./assets/profile.svg",
+        role: "user",
+    },
+    ];
+
+    const blogs: Blog[] = [
```
Comment (Segnicho) [inline aait/web/group-3/starter_project_group3/app/blogs/page.tsx:29]: the same here. factor out this data to some other folder.

**Review by Segnicho — CHANGES_REQUESTED** [review]
Looks nice left you some comments  I need you to change before merging.

File: aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx (hunk 1 above)
Comment (yohannes-07) [reply aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx]: Fixed

File: aait/web/group-3/starter_project_group3/app/blogs/page.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import React, { useState } from "react";
+import { Blog, Author } from "@/types/blog/blog";
+import BlogCard from "@/components/blog/BlogCard";
+import SearchBar from "@/components/blog/SearchBar";
+import Pagination from "@/components/blog/Pagination";
+
+const itemsPerPage = 2;
+
+const BlogPage: React.FC = () => {
+    const authors: Author[] = [
```
[the remark this answers is not in the document]
Comment (yohannes-07) [reply aait/web/group-3/starter_project_group3/app/blogs/page.tsx]: Put inside  data/blog/blog.ts file. Will be replaced with real data

File: aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx (hunk 1 above)
Comment (yohannes-07) [reply aait/web/group-3/starter_project_group3/components/blog/Pagination.tsx]: Put inside data/blog/blog.ts file. Will be replaced with real data

File: aait/web/group-3/starter_project_group3/app/blogs/page.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Segnicho) [reply aait/web/group-3/starter_project_group3/app/blogs/page.tsx]: what if in the future the user(author) data is needed? I think it is better if it is put in a dedicated file like data/blog/author.ts

**Review by Segnicho — APPROVED** [review]
LGTM

**Review by kidus101 — APPROVED** [review]

**Review by BrookW-83 — APPROVED** [review]

**Pull request merged**